zulipterminal/core.py

```diff
@@ -162,19 +162,22 @@ def maximum_popup_dimensions(self) -> Tuple[int, int]:
         return max_cols, max_rows
 
     def show_pop_up(self, to_show: Any) -> None:
-        double_lines = dict(tlcorner='╔', tline='═', trcorner='╗',
-                            rline='║', lline='║',
-                            blcorner='╚', bline='═', brcorner='╝')
+        border_lines = dict(tlcorner='▃', tline='▃', trcorner='▃',
+                            rline='▐', lline='▌',
+                            blcorner='▙', bline='▄', brcorner='▟')
+        text = urwid.Text(to_show.title, align='center')
         self.loop.widget = urwid.Overlay(
-            urwid.LineBox(to_show,
-                          to_show.title,
-                          **double_lines),
-            self.view,
-            align='center',
-            valign='middle',
-            # +2 to both of the following, due to LineBox
-            width=to_show.width + 2,
-            height=to_show.height + 2,
+                urwid.AttrMap(urwid.LineBox(urwid.Frame(
+                    header=urwid.BoxAdapter(urwid.AttrMap(
+                        urwid.Filler(text), 'popup_title'), height=1),
+                    body=to_show),
+                              **border_lines), 'popup_border'),
```

Comment:
This could be expressed a lot clearer. I know it may change if we move widgets around, but this is quite difficult to follow. Intermediate variables would help.